core/domain/stats_jobs_one_off.py

```diff
@@ -127,3 +143,2369 @@ def reduce(key, stringified_values):
                     'all: %s sum:%s' % (
                         key, state_name, all_state_hit[state_name],
                         sum_state_hit[state_name]),)
+
+
+class AnswersAudit(jobs.BaseMapReduceJobManager):
+
+    # pylint: disable=invalid-name
+    _STATE_COUNTER_ERROR_KEY = 'State Counter ERROR'
+    _UNKNOWN_HANDLER_NAME_COUNTER_KEY = 'UnknownHandlerCounter'
+    _SUBMIT_HANDLER_NAME_COUNTER_KEY = 'SubmitHandlerCounter'
+    _HANDLER_FUZZY_RULE_COUNTER_KEY = 'FuzzyRuleCounter'
+    _HANDLER_DEFAULT_RULE_COUNTER_KEY = 'DefaultRuleCounter'
+    _HANDLER_STANDARD_RULE_COUNTER_KEY = 'StandardRuleCounter'
+    _STANDARD_RULE_SUBMISSION_COUNTER_KEY = 'StandardRuleSubmitCounter'
+    _HANDLER_ERROR_RULE_COUNTER_KEY = 'ErrorRuleCounter'
+    _UNIQUE_ANSWER_COUNTER_KEY = 'UniqueAnswerCounter'
+    _CUMULATIVE_ANSWER_COUNTER_KEY = 'CumulativeAnswerCounter'
+
+    @classmethod
+    def _get_consecutive_dot_count(cls, string, idx):
+        for i in range(idx, len(string)):
+            if string[i] != '.':
+                return i - idx
+        return 0
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [stats_models.StateRuleAnswerLogModel]
+
+    @staticmethod
+    def map(item):
+        item_id = item.id
+        if 'submit' not in item_id:
+            yield (AnswersAudit._UNKNOWN_HANDLER_NAME_COUNTER_KEY, {
+                'reduce_type': AnswersAudit._UNKNOWN_HANDLER_NAME_COUNTER_KEY,
+                'rule_spec_str': item.id
+            })
+            return
+
+        period_idx = item_id.index('submit')
```

Comment:
State names can have periods. This code's gone through several iterations, but we now have a general purpose helper to make this less painful.